bin/generate-groups-inventory.bash

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+
+here="$(dirname "$(readlink -f "$0")")"
+# shellcheck source=bin/common.bash
+source "${here}/common.bash"
+
+# shellcheck source=bin/common.bash
+source "${here}/inventory-parser.bash"
+
+declare -A KUBE
+KUBE[sc]="${CK8S_CONFIG_PATH:-}/.state/kube_config_sc.yaml"
+KUBE[wc]="${CK8S_CONFIG_PATH:-}/.state/kube_config_wc.yaml"
+
+ops_kubectl() { # <prefix> <args...>
+    case "${1}" in
+        sc) kubeconfig="${KUBE[sc]}" ;;
+        wc) kubeconfig="${KUBE[wc]}" ;;
+    esac
+
+    shift
+    with_kubeconfig "$kubeconfig" kubectl "${@}"
+}
+
+
+if [ -e "${config[groups_inventory_file]}" ]; then 
+    log_info "Using ${config[groups_inventory_file]} .."
+else 
+    log_info "Creating ${config[groups_inventory_file]} .."
+    cp "${config[inventory_file]}" "${config[groups_inventory_file]}"
+fi
+
+
+# Group Elastisys Nodes
```

Comment:
Would it be possible to make it find the groups automatically? 
So it finds control plane, elastitys etc.
For nodes with the label `elastisys.io/ams-cluster-name` you can do you cluster grouping.

So that we do not need to remember to add more specific groups here in the future.

Also i guess all nodes with out a label should be grouped up into a group?

Reply:
I honestly not sure If I get your point, I think the script will find the groups automatically, for example for elastisys nodes, it will check if there are any, if not we won't have any group of that sort in the new inventory ( same for AMS clusters .. ) 

Also, I can create a dedicated group for the controle plane, but not sure how the new inventory would affect the upgrade playbook, does kubespray expect a specific group name in the inventory for the controle plane  nodes ( ie: kube_controle_plane & kube_node ), currently I'm creating the new inventory based on the inventory.ini used to install kubespray, and later adding the groups for different type of nodes

Reply:
I mean that it also find the group `elastisys_node` automatically, etc. 
So that other specific groups that we or anyone else creates in the future is picked. 

It should be fine with a `controle_plane` group.

Reply:
Okay, I maybe misunderstood the task then. I thought it was going to generate a inventory file from the cluster and then do the upgrade. 

I would say look at adding labels/taints in the tfvars when creating new clusters.
I just not a fan of hard coding to our current labels.

Reply:
Yeah, now I haven't looked looked through the scripts to much but I do agree that we shouldn't hard code it, and that why I don't want it to look for our labels either.

Having a configurable list that takes labels+taints and a regex for node names should be enough to create a mapping from nodes to groups.

Reply:
@robinAwallace Actually I understand it as you described, the groups are generated from the actual environment

Reply:
I guess it's best if we skip my idea for now and focus on generating the inventory from the state of the cluster.

@ayoubeddafali You have hard coded the labels that you check on for groups and I would like that to configurable. Either like `group_label_primary: elastisys.io/node-type` and `group_label_secondary: elastisys.io/ams-cluster-name`, or in a list `group_labels: [elastisys.io/node-type, elastisys.io/ams-cluster-name]`. You can set this in the [new general file](https://github.com/elastisys/compliantkubernetes-kubespray/blob/main/config/common/group_vars/all/ck8s-kubespray-general.yaml)